code/go/0chain.net/smartcontract/minersc/miner_test.go

```diff
@@ -0,0 +1,298 @@
+	t.Run("get sharder staker interest", func(t *testing.T) {
+		var gn, err = msc.getGlobalNode(balances)
+		require.NoError(t, err, "getting global node")
+		interest = state.Balance(float64(stakeVal) * gn.InterestRate)
+
+	})
+	return interest
+}
+
+func getMinerStakerReward(t *testing.T, balances *testBalances, msc *MinerSmartContract, miner *miner, roundsPaid int) state.Balance {
+	var rewards state.Balance
+	t.Run("get miner staker reward", func(t *testing.T) {
+		var gn, err = msc.getGlobalNode(balances)
+		require.NoError(t, err, "getting global node")
+		var mn *MinerNode
+		mn, err = msc.getMinerNode(miner.miner.id, balances)
+		require.NoError(t, err, "getting miner node")
+		blockReward := state.Balance(float64(gn.BlockReward) * gn.RewardRate)
+		minerSplit := state.Balance(float64(blockReward) * (gn.ShareRatio))
+		stakersSplit := minerSplit - state.Balance(float64(minerSplit)*mn.ServiceCharge)
+		rewards = state.Balance(float64(stakersSplit)/float64(len(miner.stakers))) * state.Balance(roundsPaid)
+	})
+	return rewards
+}
+
+func getMinerStakerInterest(t *testing.T, balances *testBalances, msc *MinerSmartContract, miner *miner) state.Balance {
+	var interest state.Balance
+	t.Run("get miner staker interest", func(t *testing.T) {
+		var gn, err = msc.getGlobalNode(balances)
+		require.NoError(t, err, "getting global node")
+		interest = state.Balance(float64(stakeVal) * gn.InterestRate)
+	})
+	return interest
+}
+
+func setupChain(t *testing.T, miners []*miner, sharders []*sharder, balances *testBalances, msc *MinerSmartContract, now int64) ([]*miner, []*sharder) {
+	miners, sharders = nil, nil
+	t.Run("add miners", func(t *testing.T) {
+		for i := 0; i < 4; i++ {
+			miners = append(miners, newMiner(t, msc, now, stakeHolders,
+				stakeVal, balances))
+			now += 10
+		}
+	})
+
+	t.Run("add sharders", func(t *testing.T) {
+		for i := 0; i < 2; i++ {
+			sharders = append(sharders, newSharder(t, msc, now, stakeHolders,
+				stakeVal, balances))
+			now += 10
+		}
+	})
+
+	msc.setDKGMinersTestHelper(t, miners, balances)
+	balances.setLFMB(createPreviousMagicBlock(miners, sharders))
+
+	t.Run("stake miners", func(t *testing.T) {
+		for _, mn := range miners {
+			for _, st := range mn.stakers {
+				_, err := st.callAddToDelegatePool(t, msc, now, stakeVal,
+					mn.miner.id, balances)
+				require.NoError(t, err, "staking miner")
+				now += 10
+			}
+		}
+	})
+
+	t.Run("stake sharders", func(t *testing.T) {
+		for _, sh := range sharders {
+			for _, st := range sh.stakers {
+				_, err := st.callAddToDelegatePool(t, msc, now, stakeVal,
+					sh.sharder.id, balances)
+				require.NoError(t, err, "staking sharder")
+				now += 10
+			}
+		}
+	})
+	return miners, sharders
+}
+
+func checkStakersBalances(t *testing.T, stakers []*Client, balances *testBalances, value state.Balance, check string) {
+	t.Run(check, func(t *testing.T) {
+		for _, staker := range stakers {
+			assert.EqualValues(t, value, balances.balances[staker.id])
```

Comment:
Tests are failing here. And use `require.EqualValues` as we want to stop failing tests.